include/deal.II/base/lazy.h

```diff
@@ -379,7 +379,8 @@ inline DEAL_II_ALWAYS_INLINE
       //
       if (!object_is_initialized.load(std::memory_order_relaxed))
         {
-          object = std::move(creator());
+          Assert(object.has_value() == false, ExcInternalError());
+          object.emplace(std::move(creator()));
```

Comment:
I changed these lines in the last two commits (which are logically separate): Instead of the move assignment operator, I use the move constructor. `std::optional::emplace()` is documented as using the move constructor to create a new object in the `std::optional`; if the `std::optional` had previously stored an object, that is first destroyed properly. We are here in a code path where the latter should not happen, and so both the old and the new code should result in the same thing: The move-creation of a new object, and no call to a destructor; the assertion checks the latter.